src/cache.rs

```diff
@@ -655,19 +663,28 @@ impl Cache {
                         },
                     );
 
-                    for id in pending {
-                        self.resolve_imports(id)?;
-                    }
+                    pending
+                        .iter()
+                        .flat_map(|id| {
+                            if let Ok(CacheOp::Done(mut ps)) = self.resolve_imports(*id) {
+                                ps.push(*id);
+                                ps
+                            } else {
+                                vec![*id]
```

Comment:
Do we really want to include `ids` where parse errors happened? I think that for the LSP we are in error-tolerant mode anyway, so this case probably doesn't happen.